pyproject.toml

```
@@ -23,3 +23,28 @@ markers = [
 [tool.black]
 target-version = ['py38', 'py39', 'py310']
 line-length = 88
+
+# https://github.com/charliermarsh/ruff
+[tool.ruff]
+line-length = 88
+target-version = "py38"
+src = ["npe2", "tests"]
+select = [
+    "E", "F", "W", #flake8
+    "UP", # pyupgrade
+    "I", # isort
+    "B", # flake8-bugbear
+    "C4", # flake8-comprehensions
+    "TID",  # flake8-tidy-imports
+    "RUF",  # ruff-specific rules
+]
```

```suggestion
]
fix = true
```

enable autofix by pre-commit ci

---

just did it in cdf23f8